src/test/ui/consts/const-eval/ub-uninhabit.stderr

```diff
@@ -1,10 +1,12 @@
-error[E0080]: this constant likely exhibits undefined behavior
+error: this constant cannot be used
   --> $DIR/ub-uninhabit.rs:18:1
    |
 LL | const BAD_BAD_BAD: Bar = unsafe { mem::transmute(()) };
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered a value of an uninhabited type
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^------------------^^^
+   |                                   |
+   |                                   entered unreachable code
    |
-   = note: The rules on what exactly is undefined behavior aren't clear, so this check might be overzealous. Please open an issue on the rust compiler repository if you believe it should not be considered undefined behavior
+   = note: #[deny(const_err)] on by default
```

Comment:
This is odd though -- so having a more precise analysis downgrades this from a hard error on an err-by-default lint? Cc @oli-obk

Reply:
This does make sense. We're now erroring during const eval instead of during validation (which happens later). We are not allowed to produce a hard error on unused constants. You'll get a hard error the moment you try to use that constant anywhere

Reply:
The original error message looks like it was a hard error on an unused constant, though?

Also I still disagree with your reading of that RFC as saying anything about hard errors in `const` definitions (as opposed to the stuff `const_prop` does); those are just totally different situations.

Reply:
oh this has nothing to do with any RFC, but with backwards compat. Unusued constants never were a hard error, so we can't make them a hard error.

Validation is new though, you didn't used to be able to cause constants that would have failed validation, so all code that can fail validation must be newer than validation. Thus validation being a hard error is not a backwards compat hazard

Reply:
Oh I see. That's odd, but it makes sense.

Could you add comments in https://github.com/rust-lang/rust/pull/53821/ explaining that?